emission/tests/netTests/extServiceTests/TestMatchStops.py

```diff
@@ -0,0 +1,314 @@
+                     out;"""
+        logging.info(f"Test query: {test_query.strip()}")
+        
+        # Generate hash the same way as in make_request_and_catch
+        query_hash = hashlib.md5(test_query.encode()).hexdigest()
+        logging.info(f"Generated query hash: {query_hash}")
+        
+        # Check that the hash is a valid string
+        self.assertIsInstance(query_hash, str, "Query hash is not a string")
+        self.assertEqual(len(query_hash), 32, "Query hash is not 32 characters (MD5 length)")
+        
+        # Create a mock cache file using JSON format
+        cache_file = os.path.join(self.cache_dir, f"{query_hash}.json")
+        cache_filename = f"{query_hash}.json"
+        logging.info(f"Creating mock cache file: {cache_file}")
+        mock_data = [{"id": 12345, "type": "node", "lat": 37.876, "lon": -122.258, "tags": {"name": "Test Stop"}}]
+        with open(cache_file, 'w') as f:
+            json.dump(mock_data, f)
+        
+        # Store the file created during this test for cleanup
+        if not hasattr(self, 'test_created_files'):
+            self.test_created_files = set()
+        self.test_created_files.add(cache_filename)
+        
+        # Create a test overpass query function to intercept the API call
+        # This lets us check if the function tries to use the cache before making an API call
+        def mock_request(*args, **kwargs):
+            logging.error("API request was made when cache should have been used")
+            self.fail("API request was made when cache should have been used")
+        
+        # Save the original function
+        original_requests = enetm.requests.post
+        
+        try:
+            # Replace with our mock
+            logging.info("Patching requests.post to detect API calls")
+            enetm.requests.post = mock_request
+            
+            # Now make the request - if cache works, it won't call our mock function
+            logging.info("Making request that should use cache")
+            result = enetm.query_overpass(test_query)
+            
+            # Verify we got a result from the cache
+            self.assertIsInstance(result, list, "Result is not a list")
+            self.assertEqual(len(result), 1, "Wrong number of results from cache")
+            self.assertEqual(result[0]['id'], 12345, "Cache returned incorrect data")
+            logging.info(f"Successfully retrieved data from cache: {result[0]} ✓")
+            
+        finally:
+            # Restore the original function
+            enetm.requests.post = original_requests
+            logging.info("Restored original requests.post function")
+            
+    def test_production_mode_bypass_cache(self):
+        """
+        Test that production mode bypasses cache.
+        
+        This test:
+        1. Makes a request to create cache files
+        2. Sets CACHE_DIR to None to simulate production mode
```

Comment:
Can you do this by setting `GEOFABRIK_OVERPASS_KEY` in os.environ instead of by setting CACHE_DIR ?
It should behave the same but I think that makes it clearer how match_stops is supposed to behave in different environments